dango/dex/src/cron.rs

```diff
@@ -955,35 +955,21 @@ fn refund_ioc_order(
 /// Updates trading volumes for both user addresses and usernames
 fn update_trading_volumes(
     storage: &mut dyn Storage,
-    oracle_querier: &mut OracleQuerier,
     account_querier: &mut AccountQuerier,
-    base_denom: &Denom,
-    filled: Udec128_6,
+    quote_denom: &Denom,
+    filled_quote: Udec128_6,
     order_user: Addr,
     volumes: &mut HashMap<Addr, Udec128_6>,
     volumes_by_username: &mut HashMap<Username, Udec128_6>,
 ) -> anyhow::Result<()> {
-    // Query the base asset's oracle price.
-    let base_asset_price = match oracle_querier.query_price(base_denom, None) {
-        Err(_err) => {
-            #[cfg(feature = "tracing")]
-            {
-                tracing::warn!(
-                    %base_denom,
-                    %_err,
-                    "Failed to query oracle price for base asset. Skipping volume update"
-                );
-            }
-
-            // If the query fails, simply do nothing and return, since we want to
-            // ensure that `cron_execute` function doesn't fail.
-            return Ok(());
-        },
-        Ok(price) => price,
-    };
+    // Only track trading volumes where quote asset is alloyed USD.
+    if quote_denom != &usd::DENOM.clone() {
+        return Ok(());
+    }
 
-    // Calculate the volume in USD for the filled order.
-    let new_volume: Udec128_6 = base_asset_price.value_of_dec_amount(filled)?;
+    // Store the volume as a decimal of whole USD amounts, not USD microunits.
+    let new_volume =
+        filled_quote.checked_div_dec_floor(Udec128::new(10).checked_pow(usd::DECIMAL)?)?;
```

Comment:
If we don't use the oracle price, then I don't see a reason to store the volume in human unit. We can just store in microunits for simplicity.

Reply:
But this means the volume data so far won't be compatible. Need to do a chain upgrade.